src/pybind/mgr/volumes/fs/operations/group.py

```diff
@@ -18,10 +18,13 @@ class Group(GroupTemplate):
     # Reserved subvolume group name which we use in paths for subvolumes
     # that are not assigned to a group (i.e. created with group=None)
     NO_GROUP_NAME = "_nogroup"
+    INTERNAL_DIRS = ["_index", "_deleting", "_legacy"]
 
     def __init__(self, fs, vol_spec, groupname):
         if groupname == Group.NO_GROUP_NAME:
             raise VolumeException(-errno.EPERM, "Operation not permitted for group '{0}' as it is an internal group.".format(groupname))
+        if groupname in Group.INTERNAL_DIRS:
```

Comment:
```
internal_groups = [Index.GROUP_NAME, Trash.GROUP_NAME, SubvolumeBase.LEGACY_CONF_DIR, Group.NO_GROUP_NAME]
if groupname in internal groups:
       raise VolumeException(-errno.EPERM, "Operation not permitted for group '{0}' as it is an internal group.".format(groupname))
```
internal_groups/internal_dirs is used by both group.py and volume.py . Maybe we can move this list to a common file? I'm not sure which one though.

Reply:
I think `fs/vol_spec.py` is a good place.

Reply:
Moved at common place `fs/vol_spec.py`.